From 2b63791ee52baa068cd7620b55897ff7d76d4577 Mon Sep 17 00:00:00 2001
From: Michael Booth <m.judd.booth@gmail.com>
Date: Mon, 13 Apr 2015 23:28:33 -0400
Subject: [PATCH 1007/1009] Change the template for group_mean_bin to set
 bounds_check and wraparound to False like all similar functions.

---
 pandas/src/generate_code.py | 3 ++-
 pandas/src/generated.pyx    | 6 ++++--
 2 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/pandas/src/generate_code.py b/pandas/src/generate_code.py
index 575fcf3..2c4217b 100644
--- a/pandas/src/generate_code.py
+++ b/pandas/src/generate_code.py
@@ -1569,7 +1569,8 @@ def group_mean_%(name)s(ndarray[%(dest_type2)s, ndim=2] out,
                 out[i, j] = sumx[i, j] / count
 """
 
-group_mean_bin_template = """
+group_mean_bin_template = """@cython.boundscheck(False)
+@cython.wraparound(False)
 def group_mean_bin_%(name)s(ndarray[%(dest_type2)s, ndim=2] out,
                    ndarray[int64_t] counts,
                    ndarray[%(dest_type2)s, ndim=2] values,
diff --git a/pandas/src/generated.pyx b/pandas/src/generated.pyx
index cab3a84..808d129 100644
--- a/pandas/src/generated.pyx
+++ b/pandas/src/generated.pyx
@@ -5648,7 +5648,8 @@ def group_mean_float32(ndarray[float32_t, ndim=2] out,
             else:
                 out[i, j] = sumx[i, j] / count
 
-
+@cython.boundscheck(False)
+@cython.wraparound(False)
 def group_mean_bin_float64(ndarray[float64_t, ndim=2] out,
                    ndarray[int64_t] counts,
                    ndarray[float64_t, ndim=2] values,
@@ -5701,7 +5702,8 @@ def group_mean_bin_float64(ndarray[float64_t, ndim=2] out,
                 out[i, j] = nan
             else:
                 out[i, j] = sumx[i, j] / count
-
+@cython.boundscheck(False)
+@cython.wraparound(False)
 def group_mean_bin_float32(ndarray[float32_t, ndim=2] out,
                    ndarray[int64_t] counts,
                    ndarray[float32_t, ndim=2] values,
-- 
2.6.0

